helix-core/src/main/java/org/apache/helix/model/LiveInstance.java

```diff
@@ -40,7 +40,20 @@
     LIVE_INSTANCE,
     ZKPROPERTYTRANSFERURL,
     RESOURCE_CAPACITY,
-    CURRENT_TASK_THREAD_POOL_SIZE
+    CURRENT_TASK_THREAD_POOL_SIZE,
+
+    /** Represents {@link LiveInstanceStatus} */
+    STATUS
+  }
+
+  /**
+   * Represents status of a live instance, eg. PAUSED.
+   */
+  public enum LiveInstanceStatus {
+    NORMAL,
+    PAUSED,
+    ENTERING_PAUSE,
+    EXITING_PAUSE
```

Comment:
Yes it could be derived. But do we want to hard code those status in the API? I thought about it. I think it's cleaner to define the status here in live instance, which could be used for controller and participant.
When exiting pause, once controller receives the exit notification, the controller status will marked as "EXITING_PAUSE" in LEADER.
So I think this enum will help.